### PR TITLE
feat: pin_derived_eph_pub+priv

### PR DESCRIPTION
**Pull request by berendsliedrecht**
Signed-off-by: Berend Sliedrecht <sliedrecht@berend.io>

### CODE REVIEW

File: apps/funke/app/crypto/bPrime.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import type { AgentContext } from '@credo-ts/core'
```
Comment (TimoGlastra) [inline apps/funke/app/crypto/bPrime.ts:1]: Everything within app is seen as routes, so I think we should add all this logic outside of `app` dir

File: packages/secure-store/kdf.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * Derive a hash from pin and salt. (which can be used to seed a key)
+ *
+ * Configuration based on recommended parameters defined in RFC 9106
+ * @see https://www.rfc-editor.org/rfc/rfc9106.html#name-parameter-choice
+ *
+ * returns a hex-encoded derived hash
+ *
+ */
+const derive = async (pin: string | Uint8Array, salt: string | Uint8Array): Promise<string> => {
+  const normalizedPin = typeof pin === 'string' ? pin : Buffer.from(pin).toString()
```
Comment (TimoGlastra) [inline packages/secure-store/kdf.ts:14]: We normally use TypedArrayEncoder

File: packages/secure-store/secure-unlock/react-native-argon2.d.ts
Comment (TimoGlastra) [inline packages/secure-store/secure-unlock/react-native-argon2.d.ts:1]: I think this needs to stay

**Review by TimoGlastra — APPROVED** [review]

File: packages/secure-store/secure-unlock/react-native-argon2.d.ts
Comment (berendsliedrecht) [reply packages/secure-store/secure-unlock/react-native-argon2.d.ts:1]: Moved it to the root of the project, made more sense to me.

File: apps/funke/app/crypto/bPrime.ts (hunk 1 above)
Comment (berendsliedrecht) [reply apps/funke/app/crypto/bPrime.ts:1]: I'll move it out, but now we just get a very messy root where `ios`, `node_modules`, `app` and now `crypto` is defined...

Comment (TimoGlastra) [reply apps/funke/app/crypto/bPrime.ts]: We can add a src directory

**Pull request merged**